src/mctpd.c

```diff
@@ -1883,6 +1886,20 @@ static int remove_peer(struct peer *peer)
 		return -EPROTO;
 	}
 
+	if (peer->pool_size > 0) {
+		for (i = 0; i < peer->pool_size; i++) {
+			eid = peer->pool_start + i;
+			if (eid < 256 && n->peers[eid]) {
```

Comment:
being a `uint8_t`, `eid` will always be < 256; the case you're looking to check here will be hidden via an overflow.

If you want to check the pool end validity, you'll need to do so in a larger type.

Reply:
Good catch! Added the check `if (peer->pool_start > 255 - i)` to avoid uint8_t overflow.